File: FractalMandelBrot/ZoomList.cpp
#include "pch.h"
#include "ZoomList.h"


namespace FractalMandelbrot
{

   ZoomList::~ZoomList()
   {
   }

   void ZoomList::Add(const Zoom & inZoom)
   {
      mZoomList.push_back(inZoom);
      mXCenter += (inZoom.x - mWidth / 2.0) * mScale;
      mYCenter += (inZoom.y - mHeight / 2.0) * mScale;
      mScale *= inZoom.scale;
   }
   std::pair <double, double> ZoomList::doZoom(int x, int y)
   {
      std::pair <double, double> lCoordinates(0.0, 0.0);
      //for (auto lZoom : mZoomList)
      {
         //lCoordinates.first = (2.0 / mWidth * (x+lZoom.x) * lZoom.scale) - 1.0 * lZoom.scale;
         //lCoordinates.second = (2.0 / mHeight * (y+lZoom.y) * lZoom.scale) - 1.0 * lZoom.scale;

         lCoordinates.first = (x - mWidth / 2)*mScale + mXCenter;
         lCoordinates.second = (y - mHeight / 2)*mScale + mYCenter;
      }
      return lCoordinates;
   }
}

File: SharpMandelbrot/ZoomList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SharpMandelbrot
{
    class ZoomList
    {
        // private:
        double mXCenter;
        double mYCenter;
        double mScale;
        int mWidth;
        int mHeight;
        List<Zoom> mZoomList;

        //public:
        public ZoomList(int pWidth = 0, int pHeight = 0)
        {
            mXCenter = 0f;
            mYCenter = 0f;
            mScale = 1.0f;
            mWidth = pWidth;
            mHeight = pHeight;
            mZoomList = new List<Zoom>();
        }

        public void Add(Zoom inZoom)
        {
            mZoomList.Add(inZoom);
            mXCenter += (inZoom.x - mWidth / 2.0) * mScale;
            mYCenter += (inZoom.y - mHeight / 2.0) * mScale;
            mScale *= inZoom.scale;
        }
        //***************************************************************************
        public Tuple<double, double> DoZoom(int x, int y)
        {
            double lX = (x - mWidth / 2) * mScale + mXCenter;
            double lY = (y - mHeight / 2) * mScale + mYCenter;
            Tuple<double, double> lCoordinates = new Tuple<double, double>(lX, lY);

            return lCoordinates;
        }

       
    };

}
